src/foam/nanos/auth/PasswordExpiryAuthService.js

```diff
@@ -22,34 +22,39 @@ foam.CLASS({
   javaImports: [
     'foam.dao.DAO',
     'foam.nanos.NanoService',
-    'static foam.mlang.MLang.EQ'
+
+    'static foam.mlang.MLang.AND',
+    'static foam.mlang.MLang.EQ',
+    'static foam.mlang.MLang.OR'
   ],
 
   methods: [
     {
       name: 'start',
       javaCode:
-`if ( getDelegate() instanceof NanoService ) {
-  ((NanoService) getDelegate()).start();
-}`
+        `if ( getDelegate() instanceof NanoService ) {
+          ((NanoService) getDelegate()).start();
+        }`
     },
     {
       name: 'login',
       javaCode:
-`User user = (User) ((DAO) getLocalUserDAO()).find(userId);
-if ( isPasswordExpired(user) ) {
-  throw new AuthenticationException("Password expired");
-}
-return getDelegate().login(x, userId, password);`
-    },
-    {
-      name: 'loginByEmail',
-      javaCode:
-`User user = (User) ((DAO) getLocalUserDAO()).inX(x).find(EQ(User.EMAIL, email.toLowerCase()));
-if ( isPasswordExpired(user) ) {
-  throw new AuthenticationException("Password expired");
-}
-return getDelegate().loginByEmail(x, email, password);`
+        `User user = (User) ((DAO) getLocalUserDAO())
+          .inX(x)
+          .find(
+            AND(
+              OR(
+                EQ(foam.nanos.auth.User.EMAIL, id.toLowerCase()),
```

Comment:
Pretty sure we don't need to include the package for `User` here and on lines 48 and 50.